cornell-dti/cu-apts
# Fix sorting on state change and horizontal scrolling

### Summary 






This pull request fixes some bugs from testing

- [x] closes issues #108 and #109 
- [x] fixed issue that makes reviews become unsorted (sorted by most recent instead of most helpful) when a helpful is added to a review and the sorting selected is most helpful
- [x] fixed horizontal scrolling on homepage

### Test Plan 
1. Go to any landlord page (eg. (`/landlord/24`). The reviews should be initially sorted by most recent. 
2. Change the sorting to most helpful. 
3. Try adding helpful to different reviews:
- if there's ties between likes, adding a helpful to one of them should move it up
- If it's not the most helpful, adding a helpful should put it in it's sorted position (eg. Most helpful is 2 and adding a helpful to one with 0 likes should put it after the one with 2)
- Sorting is unstable: two reviews that have the same number of likes can appear in any order)



### Notes 
-  I used the `overflow-x: hidden` css property to fix scrolling because there is no overflow. If there was overflow, this would hide anything that is overflowing, which isn't a problem because nothing is currently overflowing

[Update: Changed implementation to get rid of useEffects based on Sam's suggestion]

- ~~There are `react-hooks/exhaustive-deps` linter warnings from using the sortReviews function in `useEffect` which uses two states (`sortBy` and `reviewData`). I separated the re-rendering for each of the two states: one useEffect that re-renders when the `reviewData` changes and another that re-renders when the `sortBy` changes. This is because I only want to destructure the array when the `sortBy` changes not when the `reviewData` changes. (When I had one useEffect with both `sortBy` and `reviewData` as dependencies, it was re-rendering too many times and didn't actually work)~~
- ~~I disabled the linter warnings because in this case I know I don't want the useEffect that handles `reviewData` change to get called when the `sortBy` changes and vice versa.~~
